src/v/pandaproxy/schema_registry/sharded_store.cc

```diff
@@ -0,0 +1,229 @@
+}
+
+ss::future<schema> sharded_store::get_schema(const schema_id& id) {
+    co_return co_await _store.invoke_on(
+      shard_for(id), _smp_opts, [id](auto& s) {
+          return s.get_schema(id).value();
+      });
+}
+
+ss::future<subject_schema> sharded_store::get_subject_schema(
+  const subject& sub, schema_version version, include_deleted inc_del) {
+    auto v_id = co_await _store.invoke_on(
+      shard_for(sub), _smp_opts, [&sub, version, inc_del](auto& s) {
+          return s.get_subject_version_id(sub, version, inc_del).value();
+      });
+    auto s = co_await get_schema(v_id.id);
+
+    co_return subject_schema{
+      .sub = sub,
+      .version = v_id.version,
+      .id = v_id.id,
+      .type = s.type,
+      .definition = std::move(s.definition),
+      .deleted = v_id.deleted};
+}
+
+ss::future<std::vector<subject>>
+sharded_store::get_subjects(include_deleted inc_del) {
+    co_return co_await _store.map_reduce0(
+      [inc_del](store& s) { return s.get_subjects(inc_del); },
+      std::vector<subject>{},
+      [](std::vector<subject> acc, std::vector<subject> subs) {
+          acc.insert(
+            acc.end(),
+            std::make_move_iterator(subs.begin()),
+            std::make_move_iterator(subs.end()));
+          return acc;
+      });
+}
+
+ss::future<std::vector<schema_version>>
+sharded_store::get_versions(const subject& sub, include_deleted inc_del) {
+    co_return co_await _store.invoke_on(
+      shard_for(sub), _smp_opts, [&sub, inc_del](auto& s) {
+          return s.get_versions(sub, inc_del).value();
+      });
+}
+
+ss::future<std::vector<schema_version>>
+sharded_store::delete_subject(const subject& sub, permanent_delete permanent) {
+    co_return co_await _store.invoke_on(
+      shard_for(sub), _smp_opts, [&sub, permanent](auto& s) {
+          return s.delete_subject(sub, permanent).value();
+      });
+}
+
+ss::future<bool> sharded_store::delete_subject_version(
+  const subject& sub,
+  schema_version version,
+  permanent_delete permanent,
+  include_deleted inc_del) {
+    co_return co_await _store.invoke_on(
+      shard_for(sub), _smp_opts, [&sub, version, permanent, inc_del](auto& s) {
+          return s.delete_subject_version(sub, version, permanent, inc_del)
+            .value();
+      });
+}
+
+ss::future<compatibility_level> sharded_store::get_compatibility() {
+    co_return _store.local().get_compatibility().value();
+}
+
+ss::future<compatibility_level>
+sharded_store::get_compatibility(const subject& sub) {
+    co_return co_await _store.invoke_on(
+      shard_for(sub), _smp_opts, [&sub](auto& s) {
+          return s.get_compatibility(sub).value();
+      });
+}
+
+ss::future<bool>
+sharded_store::set_compatibility(compatibility_level compatibility) {
+    co_return co_await _store.map_reduce0(
+      [compatibility](store& s) {
+          return s.set_compatibility(compatibility).value();
+      },
+      true,
+      std::logical_and<>{});
```

Comment:
🔥

Reply:
It can't even fail, so it's superfluous ceremony.